packages/typescript-estree/src/parseSettings/getProjectConfigFiles.ts

```diff
@@ -0,0 +1,63 @@
+import debug from 'debug';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import type { ParseSettings } from '.';
+
+const log = debug('typescript-eslint:typescript-estree:getProjectConfigFiles');
+
+const tsconfigMatchCache = new Map<string, string | undefined>();
+
+/**
+ * @remarks Only use this for tests!
+ */
+export function clearMatchCacheForTests(): void {
+  tsconfigMatchCache.clear();
+}
+
+/**
+ * Checks for a matching TSConfig to a file including its parent directories,
+ * permanently caching results under each directory it checks.
+ *
+ * @remarks
+ * We don't (yet!) have a way to attach file watchers on disk, but still need to
+ * cache file checks for rapid subsequent calls to fs.existsSync. See discussion
+ * in https://github.com/typescript-eslint/typescript-eslint/issues/101.
+ */
+export function getProjectConfigFiles(
+  parseSettings: Pick<ParseSettings, 'filePath' | 'tsconfigRootDir'>,
+  project: string | string[] | true | undefined,
+): string | string[] | undefined {
+  if (project !== true) {
+    return project;
+  }
```

Comment:
do we want to just normalise this function to return `string[] | undefined`?

```suggestion
): string | string[] | undefined {
  if (project !== true) {
    return Array.isArray(project) ? project : [project];
  }
```